src/main/java/com/google/api/codegen/ProtocGeneratorMain.java

```diff
@@ -45,7 +46,40 @@
   private static final ArtifactType DEFAULT_ARTIFACT_TYPE = GAPIC_CODE;
 
   public static void main(String[] args) {
-    // TODO(andrealin): Implement. Handle I/O interfacing with protoc.
+    System.err.println("Parsing System.in for CodeGeneratorRequest.");
+
+    CodeGeneratorResponse response;
+    CodeGeneratorRequest request;
+    int exitCode = 0;
+
+    try {
+      request = PluginProtos.CodeGeneratorRequest.parseFrom(System.in);
+    } catch (IOException e) {
+      System.err.println("Unable to parse CodeGeneraterRequest from stdin.");
+      System.exit(1);
+      return;
+    }
+
+    try {
+      response = generate(request);
+    } catch (Exception e) {
+      StringWriter sw = new StringWriter();
+      PrintWriter pw = new PrintWriter(sw);
+      e.printStackTrace(pw);
+      pw.flush();
+      response = PluginProtos.CodeGeneratorResponse.newBuilder().setError(sw.toString()).build();
+      exitCode = 1;
+    }
+
+    try {
+      response.writeTo(System.out);
+    } catch (IOException e) {
```

Comment:
Re error handling in this function, it looks like the structure is:
 - for IO errors, write to System.err and exit immediately
 - for errors in the generate function, don't write to System.error but set the error field on the response

Is that right? If so, should we call System.exit when we get an error calling response.writeTo?

Reply:
That makes sense. Done.